api/metrics/src/main/java/io/opentelemetry/api/metrics/DoubleGaugeBuilder.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.metrics;
+
+import java.util.function.Consumer;
+
+/** A builder for Gauge metric types. These can only be asynchronously collected. */
+public interface DoubleGaugeBuilder {
+  /**
+   * Sets the description for this instrument.
+   *
+   * <p>Description strings should follow the instrument description rules:
+   * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-description
+   */
+  DoubleGaugeBuilder setDescription(String description);
+
+  /**
+   * Sets the unit of measure for this instrument.
+   *
+   * <p>Unit strings should follow the instrument unit rules:
+   * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-unit
+   */
+  DoubleGaugeBuilder setUnit(String unit);
+
+  /** Sets the gauge for recording {@code long} values. */
+  LongGaugeBuilder ofLongs();
```

Comment:
I don't think this is a very "java" pattern.

Reply:
To be fair, I am a Scala guy, so it's not surprising.  However, a few Java-isms I based this on (again as a Scala outsider when investigating boxing + specialization):

- https://docs.oracle.com/javase/8/docs/api/java/util/PrimitiveIterator.OfLong.html
- https://beam.apache.org/releases/javadoc/2.0.0/org/apache/beam/sdk/transforms/Sum.html#ofLongs--

TL;DR: it was the JDK's chosen name when you specialize an interface for primitive inputs, so I chose it.  If you have an alternative suggestion, happy to hear it.

Reply:
I know this is related to DateTimes, but provides context for the conversation: https://docs.oracle.com/javase/tutorial/datetime/overview/naming.html